From 5018f81c4733c3fd638cfa3aff5f1f8e5ab82cdf Mon Sep 17 00:00:00 2001
From: letsfindaway <me@letsfindaway.de>
Date: Wed, 16 Nov 2022 08:31:29 +0100
Subject: [PATCH 08/10] feat: disable software update check

- for packages maintained by a Linux distribution,
  a software update check is not relevant
- disable the check in the settings
- hide the associated checkbox in the preferences
---
 resources/etc/OpenBoard.config       | 5 ++---
 src/core/UBPreferencesController.cpp | 1 +
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/resources/etc/OpenBoard.config b/resources/etc/OpenBoard.config
index 89f3a72e..3ecf4e98 100644
--- a/resources/etc/OpenBoard.config
+++ b/resources/etc/OpenBoard.config
@@ -1,9 +1,8 @@
 [App]
 AngleTolerance=4
-HideCheckForSoftwareUpdate=false
+HideCheckForSoftwareUpdate=true
 HideSwapDisplayScreens=true
-EnableAutomaticSoftwareUpdates=true
-EnableSoftwareUpdates=true
+EnableAutomaticSoftwareUpdates=false
 EnableStartupHints=true
 FavoriteToolURIs=openboardtool://openboard/mask, openboardtool://ruler, openboardtool://compass, openboardtool://protractor, openboardtool://triangle, openboardtool://magnifier, openboardtool://cache
 IsInSoftwareUpdateProcess=false
diff --git a/src/core/UBPreferencesController.cpp b/src/core/UBPreferencesController.cpp
index c1ee6614..a60d10c2 100644
--- a/src/core/UBPreferencesController.cpp
+++ b/src/core/UBPreferencesController.cpp
@@ -85,6 +85,7 @@ UBPreferencesController::UBPreferencesController(QWidget *parent)
     mPreferencesWindow = new UBPreferencesDialog(this,parent, Qt::Dialog);
     mPreferencesUI = new Ui::preferencesDialog();  // deleted in
     mPreferencesUI->setupUi(mPreferencesWindow);
+    mPreferencesUI->softwareUpdateGroupBox->hide();     // disable check for software update
     adjustScreens(1);
     connect(mDesktop, &QDesktopWidget::screenCountChanged, this, &UBPreferencesController::adjustScreens);
 
-- 
2.40.1

